catalog/src/main/java/org/killbill/billing/catalog/api/user/DefaultCatalogUserApi.java

```diff
@@ -224,5 +220,23 @@ private InternalTenantContext createInternalTenantContext(final TenantContext te
         return internalCallContextFactory.createInternalTenantContextWithoutAccountRecordId(tenantContext);
     }
 
+    private ValidationErrors validateCatalogInternal(final String catalogXML, final InternalTenantContext internalTenantContext) throws ValidationException, JAXBException, CatalogApiException, TransformerException, SAXException, IOException {
```

Comment:
Instead of throwing here, could you extract error messages from the exception and add them into the errors?

I'm thinking specifically of errors like:

```
				Entity: line 2: parser error : Opening and ending tag mismatch: effectiveDate line 2 and effectiveate
				    <effectiveDate>2013-02-08T00:00:00+00:00</effectiveate>
				                                                           ^
				Entity: line 4: parser error : Opening and ending tag mismatch: recurringBillingMode line 4 and recurringBillingMde
				    <recurringBillingMode>IN_ADVANCE</recurringBillingMde>
				                                                          ^
				Entity: line 10: parser error : Opening and ending tag mismatch: category line 10 and catgory
				            <category>BASE</catgory>
```

or mis-matches in the references (e.g. pricelist with a plan that isn't defined):

```
Caused by: org.xml.sax.SAXParseException; lineNumber: 52; columnNumber: 11; cvc-id.1: There is no ID/IDREF binding for IDREF 'Standard'.
```

Reply:
Fixed as follows:

- added `e.getLinkedException().getMessage()` for `JAXBException` (Since the error message is present in the `linkedException`)
- Added `throw new IllegalStateException(e)` for `TransformerException/IOException/SAXException` to keep this consistent with the earlier `uploadCatalog` code

Let me know how this looks.